src/components/Asset/AssetActions/Trade/Output.tsx

```diff
@@ -4,93 +4,90 @@ import { useAsset } from '@context/Asset'
-            ? new Decimal(swapFee).mul(new Decimal(values.baseToken))
-            : 0
-          : isValidNumber(swapFee) && isValidNumber(values.datatoken)
-          ? new Decimal(swapFee).mul(new Decimal(values.datatoken))
-          : 0
-      setSwapFeeValue(value.toString())
-    }
-    getSwapFee()
-  }, [poolAddress, values, isAssetNetwork, swapFee])
-
   // Get output values
   useEffect(() => {
     if (!poolAddress || !isAssetNetwork) return
 
     async function getOutput() {
-      // Minimum received
-      // TODO: check if this here is redundant cause we call some of that already in Swap.tsx
-      const maxImpact = 1 - Number(values.slippage) / 100
-      const maxPrice =
-        values.type === 'buy'
-          ? isValidNumber(values.datatoken) && isValidNumber(maxImpact)
-            ? new Decimal(values.datatoken)
-                .mul(new Decimal(maxImpact))
-                .toString()
-            : '0'
-          : isValidNumber(values.baseToken) && isValidNumber(maxImpact)
-          ? new Decimal(values.baseToken).mul(new Decimal(maxImpact)).toString()
-          : '0'
+      if (!values.baseToken || !values.datatoken || !values.output) return
+
+      const output =
+        values.output === 'exactIn'
+          ? new Decimal(
+              values.type === 'sell' ? values.baseToken : values.datatoken
+            )
+              .mul(
+                new Decimal(1)
+                  .minus(new Decimal(values.slippage).div(new Decimal(100)))
+                  .toString()
+              )
+              .toString()
+          : new Decimal(
+              values.type === 'sell' ? values.datatoken : values.baseToken
+            )
+              .mul(
+                new Decimal(1)
+                  .plus(new Decimal(values.slippage).div(new Decimal(100)))
+                  .toString()
+              )
+              .toString()
 
-      setMaxOutput(maxPrice)
+      setOutputWithSlippage(output)
     }
     getOutput()
   }, [poolAddress, values, isAssetNetwork])
 
   return (
     <div className={styles.output}>
       <div>
-        <p>Minimum Received</p>
+        <p>
+          {values.output === 'exactIn' ? 'Minimum Received' : 'Maximum Sent'}
+        </p>
         <Token
-          symbol={values.type === 'buy' ? dtSymbol : oceanSymbol}
-          balance={maxOutput}
+          symbol={
+            values.type === 'buy'
+              ? values.output === 'exactIn'
+                ? poolInfo.datatokenSymbol
+                : poolInfo.baseTokenSymbol
+              : values.output === 'exactIn'
+              ? poolInfo.baseTokenSymbol
+              : poolInfo.datatokenSymbol
+          }
+          balance={outputWithSlippage}
         />
       </div>
       <div>
         <p>Swap fee</p>
         <Token
-          symbol={`${values.type === 'buy' ? oceanSymbol : dtSymbol} ${
-            swapFee ? `(${swapFee}%)` : ''
+          symbol={`${
```

Comment:
consistency, don't need to stringify this and just output values like above

Reply:
we kinda do, we have 2 strings here, above it's just one logical condition